crowdaction/main.go

```diff
@@ -0,0 +1,294 @@
+		}
+		//get items for a partition key
+		result, err := getItems(pk, sk)
+		//error handling https://play.golang.org/p/D5xTeZP9VnU
+		if err != nil {
+			mapErr[pk] = err
+		}
+		if result != nil {
+			//add partition key result to the total result for the date range
+			items = append(items, result)
+		}
+	}
+
+	if len(mapErr) > 0 {
+		e := pkError{}
+		for k, v := range mapErr {
+			e.Pk = k
+			e.Err = v.Error()
+			jsonErr, err := json.Marshal(e)
+			if err != nil {
+				errors = append(errors, err.Error())
+			} else {
+				errors = append(errors, string(jsonErr))
+			}
+		}
+
+		body := "[" + strings.Join(errors, ",") + "]"
+		return events.APIGatewayProxyResponse{Body: body, StatusCode: http.StatusInternalServerError}, nil
+
+	}
+
+	if items == nil {
+		body, _ := json.Marshal(map[string]string{"message": "no active crowdactions found"})
+		return events.APIGatewayProxyResponse{
+			Body:       string(body),
+			StatusCode: http.StatusNotFound,
+		}, nil
+	}
+
+	//no error yet if we are here
+	var listCrowdaction []string
+	for _, v := range items {
+		for _, item := range v.Items {
+			err := dynamodbattribute.UnmarshalMap(item, &crowdaction)
+			if err != nil {
+				return events.APIGatewayProxyResponse{Body: err.Error(), StatusCode: http.StatusBadRequest}, nil
+			}
+			action, err := json.Marshal(map[string]interface{}{"data": crowdaction})
+			if err != nil {
+				return events.APIGatewayProxyResponse{Body: err.Error(), StatusCode: http.StatusBadRequest}, nil
+			}
+			listCrowdaction = append(listCrowdaction, string(action))
+		}
+	}
+	body := "[" + strings.Join(listCrowdaction, ",") + "]"
+	return events.APIGatewayProxyResponse{
+		Body:       body,
+		StatusCode: http.StatusOK,
+	}, nil
+}
+
+//get details about a crowd action
+func getCrowdaction(crowdactionID string, req events.APIGatewayV2HTTPRequest) (events.APIGatewayProxyResponse, error) {
+
+	var crowdaction Crowdaction
+
+	val := prefixPKcrowdactionID + crowdactionID
+	out, err := getItem(val)
+
+	if err != nil {
+		body, err := json.Marshal(map[string]interface{}{"message": err.Error()})
+		if err != nil {
+			return events.APIGatewayProxyResponse{Body: err.Error(), StatusCode: http.StatusBadRequest}, nil
+		}
+		return events.APIGatewayProxyResponse{
+			Body:       string(body),
+			StatusCode: http.StatusInternalServerError,
+		}, nil
+	}
+
+	if out.Item == nil {
+		body, err := json.Marshal(map[string]string{"message": "crowdaction does not exist"})
+		if err != nil {
+			return events.APIGatewayProxyResponse{Body: err.Error(), StatusCode: http.StatusBadRequest}, nil
+		}
+		return events.APIGatewayProxyResponse{
+			Body:       string(body),
+			StatusCode: http.StatusNotFound,
+		}, nil
+	}
+
+	err = dynamodbattribute.UnmarshalMap(out.Item, &crowdaction)
+	if err != nil {
+		return events.APIGatewayProxyResponse{Body: err.Error(), StatusCode: http.StatusBadRequest}, nil
+	}
+
+	body, err := json.Marshal(map[string]interface{}{"data": crowdaction})
```

Comment:
Not all fields of the crowdaction should be returned!
(Exclude `Crowdaction.Code`)